lib/alghom.gi

```diff
@@ -906,7 +906,7 @@ PreImagesRepresentativeOperationAlgebraHomomorphism := function( ophom, mat )
     return mat;
 end;
 
-InstallMethod( PreImagesRepresentativeNC,
+InstallMethod( PreImagesRepresentative,
```

Comment:
Is the idea here that these methods already do all appropriate checking, so we don't need to attach "NC" to them?

Reply:
The reason I ask is that there are two possible issues this raises:

1) We should probably have tests which make sure that make sure this is a "non-NC" method.

2) Installing a method for PreImagesRepresentative, but not for PreImagesRepresentativeNC, means there could be cases where PreImageRep works, while PreImageRepNC doesn't.

I realise that these two things are currently issues, as PreImageRep and PreImageRepNC are the same operation with different names, but it's probably worth having a long term plan for handling this before installing methods for both PreImagesRep and PreImagesRepNC in different places.

Reply:
Surely all methods, such as in alghom.gi, should be for PreImagesRepresentativeNC, PreImagesSetNC and PreImagesElmNC.  The real question is whether calls to such methods should be to the NC or non-NC versions.  Since, up until now, there has been no test that element elm in in the image, it is reasonable to assume that the authors of such calls were confident that this was the case.  That is why all the calls have been changed to the NC versions.  In due course, once packages have had the chance to make changes, non-NC methods, making a check, will be added.  Once that is done the status of these calls could be reviewed.